csclient/params/params.go

```diff
@@ -54,9 +54,15 @@ const (
 // StatsUpdateRequest holds the parameters for a put to /stats/update.
 // See https://github.com/juju/charmstore/blob/v4/docs/API.md#stats-update
 type StatsUpdateRequest struct {
-	Timestamp      time.Time
-	Type           string
-	CharmReference *charm.Reference
+	Entries []StatsUpdateEntry
+}
+
+// StatsUpdateEntry holds an entry of the StatsUpdateRequest for a put to /stats/update.
+// See https://github.com/juju/charmstore/blob/v4/docs/API.md#stats-update
+type StatsUpdateEntry struct {
+	Timestamp      time.Time        // Time when the update did happen.
+	Type           string           // One of the constant Download, Traffic or Deploy.
```

Comment:
Maybe define this as a type:

e.g.

    type StatsUpdateType string
    const (
        UpdateDownload StatsUpdateType = "download"
        UpdateTraffic StatsUpdateType = "traffic"
        UpdateDeploy StatsUpdateType = "deploy"
     )

?

Then the possible values become more obvious and are easily commented themselves, etc.